interfaces/repo.go

```diff
@@ -471,9 +471,14 @@ func (r *Repository) securitySnippetsForSnap(snapName string, securitySystem Sec
 			return nil, err
 		}
 		if snippet != nil {
-			for appName := range slot.Apps {
-				securityTag := snap.AppSecurityTag(snapName, appName)
+			if len(slot.Apps) == 0 {
```

Comment:
The problem and code path here is not udev specific. It just happens that udev is the only one exercising this particular case today.